decorated_kanon.js

```diff
@@ -0,0 +1,172 @@
+/* "Decorated Kan-on" romanization
+ * - vowels based on romanized Kan-on
+ * - consonants based on romanized Go-on
+ * - distinctions not made in onyomi are marked with diacritics and digraphs
+ * More details at https://kanjisense.com/dict/middle-chinese-pronunciation
+ * @author justinsilvestre
+ */
+
+if (!音韻地位) return [
+  ['聲調', [1, '平:∅ 上:ˬ 去:ˎ 入:∅', '平:1 上:2 去:3 入:4']],
+];
+const toneDiacritics = 音韻地位.聲調 === '平:∅ 上:ˬ 去:ˎ 入:∅';
+const is = (...x) => 音韻地位.屬於(...x);
+
+let 聲母 = {
+  幫: 'p',   滂: 'pʻ',   並: 'b',   明: 'm',
+  端: 't',   透: 'tʻ',   定: 'd',   泥: 'n',  來: 'l',
+  知: 't',  徹: 'tʻ',  澄: 'd',  孃: 'n',
+  精: 'ts',  清: 'tsʻ',  從: 'dz',                     心: 's',  邪: 'z',
+  莊: 'tṣ', 初: 'tṣʻ', 崇: 'dẓ',                    生: 'ṣ', 俟: 'ẓ',
+  章: 'tś', 昌: 'tśʻ', 常: 'dź', 日: 'nź',          書: 'ś', 船: 'ź', 以: 'y',
+  見: 'k',   溪: 'kʻ',   羣: 'g',   疑: 'ng',
+  影: "ʾ",   曉: 'kh',    匣: 'gh',                                          云: '',
+}[音韻地位.母];
+
+let get韻母 = {
+  // 一等韻
+  東: () => {
+    if (is`入聲 三等`) {
+      if (is`幫組`) return 'ūk';
+      if (is`以母`) return 'ẁīk';
+      return is`莊組 或 章組` ? 'yūk' : 'wīk';
+    }
+    if (is`三等`) {
+      return is`幫組` ? 'ūng' : 'iūng';
+    }
+    return is`影母` ? 'wōng' : 'ōng';
+  },
+  冬: 'ong',
+  模: () => is`影母` ? 'wo' : 'o',
+  泰: () => is`合口` ? 'wāi' : 'āi',
+  灰: 'wai',
+  咍: 'ai',
+  魂: 'won',
+  痕: 'on',
+  寒: () => is`合口` ? 'wan' : 'an',
+  豪: 'au',
+  歌: () => {
+    if (is`合口`) return is`三等` ? 'wȧ' : 'wa';
+    return is`三等` ? 'ya' : 'a';
+  },
+  唐: () => is`合口` ? 'wang' : 'ang',
+  登: () => is`合口` ? 'wŏng' : 'ŏng',
+  侯: 'ou',
+  覃: 'am',
+  談: 'ām',
+
+  // 二等韻
+  江: 'ạ̊ng',
+  佳: () => is`合口` ? 'wạ̈' : 'ạ̈',
+  皆: () => is`合口` ? 'wạ̈i' : 'ạ̈i',
+  夬: () => is`合口` ? 'wạ' : 'ạ',
+  刪: () => is`合口` ? 'wạn' : 'ạn',
+  山: () => is`合口` ? 'wạ̈n' : 'ạ̈n',
+  肴: 'ạu',
+  麻: () => {
+    if (is`三等`) return 'yạ';
+    return is`合口` ? 'wạ' : 'ạ';
+  },
+  庚: () => {
+    if (is`三等`) return is`合口` ? 'wẹng' : 'ẹng';
+    return is`合口` ? 'wạng' : 'ạng';
+  },
+  耕: () => is`合口` ? 'wạ̈ng' : 'ạ̈ng',
+  咸: 'ạ̈m',
+  銜: 'ạm',
+
+  // 四等韻
+  齊: () => is`合口` ? 'wei' : 'ei',
+  先: () => is`合口` ? 'wen' : 'en',
+  蕭: 'eu',
+  青: () => is`合口` ? 'weng' : 'eng',
+  添: 'em',
+
+  // 三等陰聲韻
+  支: () => {
+    if (is`合口 非(見組 或 影組 或 以母)`) return 'uï';
+    if (is`合口`) return is`以母 或 (重紐A類 (幫組 或 見組 或 影組))` ? 'ẁï' : 'wï';
+    return is`以母 或 (重紐A類 (幫組 或 見組 或 影組))` ? 'yï' : 'ï';
```

Comment:
```suggestion
    if (is`合口 非(見影組 或 以母)`) return 'uï';
    if (is`合口`) return is`以母 或 重紐A類 鈍音` ? 'ẁï' : 'wï';
    return is`以母 或 重紐A類 鈍音` ? 'yï' : 'ï';
```
and so on
`鈍音` can actually be omitted, IIRC

Reply:
@ayaka14732